overlord/servicestate/quota_handlers_test.go

```diff
@@ -1027,11 +1023,14 @@ func (s *quotaHandlersSuite) TestQuotaSnapMixSnapsWithSubgroupsHappy(c *C) {
 func (s *quotaHandlersSuite) TestQuotaSnapFailToMixSubgroupWithServices(c *C) {
 	r := s.mockSystemctlCalls(c, join(
 		[]expectedSystemctl{{expArgs: []string{"daemon-reload"}}},
-
 		// CreateQuota for foo
 		systemctlCallsForSliceStart("foo"),
+		// CreateQuota for foo2 - foo has changed
+		systemctlCallsForServiceRestart("test-snap"),
 
+		[]expectedSystemctl{{expArgs: []string{"daemon-reload"}}},
```

Comment:
I'm kind of confused, this test is called TestQuotaSnapFailToMixSubgroupWithServices, should it just continue to fail? I'm missing where is that change that puts a service in a group? do we have preexisting tests that were doing nothing, if we did that and I missed it is kind of a bad idea... I'm very confused honestly by the changes to this test file. As I don't see new happy tests. Just fail tests that now do more??

Reply:
You are misunderstanding my comment, last PR (the first related to per-service quota), this unit test was introduced, that puts services into a sub-group. The issue was that this does not trigger a daemon-reload or the correct unit changes. This PR fixes this and because of this existinging unit tests need modifications. This unit test needs to continue failing, but the expected calls have changed to now be correct.

Reply:
And to be clear, this unit test was doing exactly as it was supposed to. The only change is that we were not observing the correct service-changes, which needs this PR to be correct. But the point of the unit test was never to test that these were correct.